leshan-core/src/test/java/org/eclipse/leshan/senml/SenMLSerializerTest.java

```diff
@@ -120,7 +124,22 @@ public void serialize_device_object() {
         b.append("{\"n\":\"13\",\"v\":1.3674912E9},");
         b.append("{\"n\":\"14\",\"vs\":\"+02:00\"},");
         b.append("{\"n\":\"15\",\"vs\":\"U\"}]");
-        
+
         Assert.assertTrue(json.equals(b.toString()));
     }
+
+    @Test
+    public void serialize_device_object_to_senml_cbor() {
+        byte[] cbor = SenMLCbor.toSenMLCbor(givenDeviceObjectInstance());
+        LOG.debug("CBOR: " + Hex.encodeHexString(cbor));
+
+        String expectedValue = "9fbf622d32652f332f302f613061306133744f70656e204d6f62696c6520416c6c69616e6365ffbf6"
+                + "13061316133764c69676874776569676874204d324d20436c69656e74ffbf61306132613369333435303030313233f"
+                + "fbf613063362f306132fa3f800000ffbf613063362f316132fa40a00000ffbf613063372f306132fa456d8000ffbf6"
+                + "13063372f316132fa459c4000ffbf613063382f306132fa42fa0000ffbf613063382f316132fa44610000ffbf61306"
+                + "1396132fa42c80000ffbf61306231306132fa41700000ffbf61306431312f306132fa00000000ffbf61306231336132"
+                + "fa4ea30485ffbf61306231346133662b30323a3030ffbf613062313561336155ffff";
```

Comment:
This doesn't look like to the [CBOR payload example](http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1-20180710-A.html#7-4-5-0-745-CBOR) from the specification ? Does it should ?

Reply:
Good point, example in TS is {-2: "/3/0/", 0: "0", 3: "Open Mobile Alliance"}...., however generated CBOR notation in test case is {"-2": "/3/0/", 0: "0", 3: "Open Mobile Alliance"}. Field name shall be number type, instead of string type. I will check with Jackson CBOR library, to correct it.